backend/de.metas.acct.base/src/main/java-legacy/org/compiere/acct/DocLine_Allocation.java

```diff
@@ -310,6 +305,16 @@ public boolean isSalesPurchaseInvoiceToCompensate(final AcctSchemaId acctSchemaI
 			return false;
 		}
 
+		// The invoice shall not be a credit memo, unless it's allocated with a service
+		if (isCreditMemoInvoice())
+		{
+			if(!counterLine.isService())
```

Comment:
why checking isService()?

isService() returns true if the product is a service product.
In  your case, there is no C_AllocationLine.M_Product_ID => returns false.
see org.compiere.acct.DocLine#isService => ... => org.compiere.acct.DocLine#checkIsItem

Reply:
True. That always returns true. Got rid of the check